source/src/ble/cs_BleCentral.cpp

```diff
@@ -27,14 +27,32 @@ void handle_discovery(ble_db_discovery_evt_t* event) {
 	BleCentral::getInstance().onDiscoveryEvent(*event);
 }
 
+#if NORDIC_SDK_VERSION > 15
+// TODO: Get rid of this macro and only create data structure if class instance is used...
+NRF_BLE_GQ_DEF(m_ble_gatt_queue,
+		NRF_SDH_BLE_PERIPHERAL_LINK_COUNT,
+		NRF_BLE_GQ_QUEUE_SIZE);
+#endif
+
 BleCentral::BleCentral() {
+#if NORDIC_SDK_VERSION > 15
+	_queue = &m_ble_gatt_queue;
+#endif
 }
 
 void BleCentral::init() {
 	_discoveryModule.discovery_in_progress = false;
-	_discoveryModule.discovery_pending = false;
 	_discoveryModule.conn_handle = BLE_CONN_HANDLE_INVALID;
+#if NORDIC_SDK_VERSION == 15
+	// assuming this is zero-initialized to start with...
+	_discoveryModule.discovery_pending = false;
 	uint32_t nrfCode = ble_db_discovery_init(handle_discovery);
+#else
+	memset(&_discoveryInit, 0, sizeof(ble_db_discovery_init_t));
```

Comment:
Is memset the way to go? There is usually a way to init the struct.

Reply:
Minor. Feel free to fix.

Reply:
Addressed by https://github.com/crownstone/bluenet/pull/137/commits/9c7ee719dfa4500664349e559d8849306c31a359